datalake-api/src/main/java/com/sequenceiq/sdx/api/model/SdxCustomClusterRequest.java

```diff
@@ -0,0 +1,59 @@
+package com.sequenceiq.sdx.api.model;
+
+import java.io.IOException;
+
+import org.apache.commons.lang3.tuple.Pair;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.request.image.ImageSettingsV4Request;
+
+public class SdxCustomClusterRequest extends SdxClusterRequestBase {
+    private ImageSettingsV4Request imageSettingsV4Request;
+
+    public ImageSettingsV4Request getImageSettingsV4Request() {
+        return imageSettingsV4Request;
+    }
+
+    public void setImageSettingsV4Request(ImageSettingsV4Request imageSettingsV4Request) {
+        this.imageSettingsV4Request = imageSettingsV4Request;
+    }
+
+    public String getRuntime() {
+        return null;
+    }
+
+    @Override
+    public String toString() {
+        String baseString = super.toString();
+        String additionalString = ", imageSettingsV4Request=" + imageSettingsV4Request +  '}';
+        String ownString = baseString.replace("SdxClusterRequestBase", "SdxCustomClusterRequest");
+        return ownString.replace("}", additionalString);
+    }
+
+    public Pair<SdxClusterRequest, ImageSettingsV4Request> convert() throws IOException {
+        SdxClusterRequest newRequest = new SdxClusterRequest();
+        newRequest.setEnvironment(getEnvironment());
+        newRequest.setClusterShape(getClusterShape());
+        newRequest.setTags(getTags());
+        newRequest.setCloudStorage(getCloudStorage());
+        newRequest.setExternalDatabase(getExternalDatabase());
+        newRequest.setAws(getAws());
+        newRequest.setEnableRangerRaz(isEnableRangerRaz());
+
+        return new Pair<SdxClusterRequest, ImageSettingsV4Request>() {
+            @Override
+            public SdxClusterRequest getLeft() {
+                return newRequest;
+            }
+
+            @Override
+            public ImageSettingsV4Request getRight() {
+                return getImageSettingsV4Request();
+            }
+
+            @Override
+            public ImageSettingsV4Request setValue(ImageSettingsV4Request value) {
+                return value;
```

Comment:
Or perhaps throw `UnsupportedOperationException`?

Reply:
I changed it to "throw exception "